providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/grpc/GrpcConnector.java

```diff
@@ -91,12 +91,13 @@ public void shutdown() throws Exception {
         try {
             if (this.channel != null) {
                 this.channel.shutdown();
-                this.channel.awaitTermination(5, TimeUnit.SECONDS);
+                this.channel.awaitTermination(this.deadline, TimeUnit.MILLISECONDS);
             }
         } finally {
             this.cache.clear();
             if (this.channel != null) {
                 this.channel.shutdownNow();
+                this.channel.awaitTermination(this.deadline, TimeUnit.MILLISECONDS);
```

Comment:
I'm using the deadline here instead of the arbitrary 5s. This makes sense, because in theory, the only thing blocking the shutdown should be in-flight calls, which will be limited to the deadline.